src/indra_cogex/sources/odinson/grammars.py

```diff
@@ -0,0 +1,20 @@
+class Rule:
+    def __init__(self, name, label, type, pattern):
+        self.name = name
+        self.label = label
+        self.type = type
+        self.pattern = pattern
```

Comment:
you should use [dataclasses](https://docs.python.org/3/library/dataclasses.html) for this, which uses type annotations to automatically write the `__init__` function for you

```suggestion
from dataclasses import dataclass
class Rule:
    name: str
    label: str
    type: str
    pattern: str
```